File: fixtures/scale/scala/large/src/cryptobench/ciphers/xorcipher/XorCipherSuite.scala
package cryptobench.ciphers.xorcipher

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class XorCipherSuite extends CipherSuite {
  override def name: String = "xorcipher"

  override def run(): SuiteResult = {
    val cipher: Cipher = new XorCipherCipher(XorCipherKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- XorCipherVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}

File: fixtures/scale/dart/large/lib/ciphers/xorcipher/xorcipher_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'xorcipher_cipher.dart';
import 'xorcipher_key.dart';
import 'xorcipher_vectors.dart';

class XorCipherSuite implements CipherSuite {
  @override
  String name() => 'xorcipher';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = XorCipherCipher(XorCipherKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in xorcipherSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
